homeassistant/components/mill/climate.py

```diff
@@ -111,23 +108,13 @@ def __init__(
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, heater.device_id)},
             manufacturer=MANUFACTURER,
-            model=f"Generation {heater.generation}",
+            model=heater.model,
             name=heater.name,
         )
-        if heater.is_gen1:
-            self._attr_hvac_modes = [HVACMode.HEAT]
-        else:
-            self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
-
-        if heater.generation < 3:
-            self._attr_supported_features = (
-                ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
-            )
-            self._attr_target_temperature_step = PRECISION_WHOLE
-        else:
-            self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
-            self._attr_target_temperature_step = PRECISION_HALVES
+        self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
 
+        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+        self._attr_target_temperature_step = PRECISION_TENTHS
```

Comment:
Is this really the case for all generations? I thought only the last (3rd) generation supported anything else than whole numbers? I only have gen 3 heaters, so I cannot test this. The display on gen 1 and 2 only show whole numbers.

My gen 3 heaters works fine with this change.

(and a side note: the code looks much cleaner with this change, so I hope this is the case and we can use `PRECISION_TENTHS` for all generations)

Reply:
Not 100% sure. 
The display on my heater only shows integers, but the app allows me to set 0.1 values.
I am not sure if the API provides info about the generation?

Reply:
If the app now uses 0.1 precision I would assume this is all good 👍